### ISSUE TITLE
Ups and downs for columns with defaults fails

### ISSUE THREAD

**Issue by Soulforged** [livehelpnow/tds_ecto#26]
Hi,

Trying this schema:

``` elixir
defmodule Repo.Migrations.AddPayloadTypeToHistoryItem do
  use Ecto.Migration

  def change do
    alter table(:history_items) do
      add :payload_type, :string, [default: "string", null: false]
    end
  end
end
```
1. Execute ecto.migrate.

The result is: 

```
Generated SQL: ALTER TABLE [history_items] ADD [payload_type] nvarchar(255) NOT NULL,
ALTER TABLE [history_items] IF (OBJECT_ID('DF_payload_type', 'D') IS NOT NULL) BEGIN ALTER TABLE [history_items] DROP CONSTRAINT DF_payload_type END; 
** (Tds.Error) 1 (3971): The server failed to resume the transaction. Desc:5700000001.
    (ecto) lib/ecto/adapters/sql.ex:544: Ecto.Adapters.SQL.transaction/3
    (ecto) lib/ecto/migrator.ex:219: anonymous fn/4 in Ecto.Migrator.migrate/4
    (elixir) lib/enum.ex:1043: anonymous fn/3 in Enum.map/2
    (elixir) lib/enum.ex:1387: Enum."-reduce/3-lists^foldl/2-0-"/3
    (elixir) lib/enum.ex:1043: Enum.map/2
    (ecto) lib/mix/tasks/ecto.rollback.ex:63: Mix.Tasks.Ecto.Rollback.run/2
    (mix) lib/mix/cli.ex:55: Mix.CLI.run_task/2
```

The line of log "Generated SQL" is my addendum.
Notice how the addition of the constraint and default value is done as a separate command, that will fail if the column has a NOT NULL restriction because it's trying to add the column before specifying the default value.

Something similar happens when rolling back that migration, the DROP statement for the column is correctly generated but the constraint is not dropped.

**Comment by amarsahinovic** [comment]
Any progress on this? I'm having the same issue when creating tables:

```
00:16:30.185 [info]  == Running Example.Repo.Migrations.CreateCertification.change/0 forward

00:16:30.185 [info]  create table certifications

00:16:30.196 [debug] CREATE TABLE [certifications] ([id] bigint NOT NULL PRIMARY KEY IDENTITY, [name] nvarchar(255) NULL, [description] nvarchar(255) NULL, [is_active] bit NULL CONSTRAINT DF_is_activ
e DEFAULT 0, [inserted_at] datetime2 NOT NULL, [updated_at] datetime2 NOT NULL) [] ERROR query=16.0ms

00:16:30.201 [info]  == Migrated in 0.1s
** (Tds.Error) 1 (3971): The server failed to resume the transaction. Desc:3600000001.
    (ecto) lib/ecto/adapters/sql.ex:560: Ecto.Adapters.SQL.transaction/3
    (ecto) lib/ecto/migrator.ex:226: anonymous fn/4 in Ecto.Migrator.migrate/4

    (elixir) lib/enum.ex:1088: Enum."-map/2-lists^map/1-0-"/2
    (ecto) lib/mix/tasks/ecto.migrate.ex:63: anonymous fn/4 in Mix.Tasks.Ecto.Migrate.run/2
    (elixir) lib/enum.ex:604: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:604: Enum.each/2
    (mix) lib/mix/cli.ex:58: Mix.CLI.run_task/2
    (elixir) lib/code.ex:363: Code.require_file/2
```

**Comment by mjaric** [comment]
Resolved in tds_ecto 2.2
